Unorsk/unique as default type modifier

unorsk opened the pull request:
## Overview

Implementation for [4539](https://github.com/unisonweb/unison/issues/4539)

## Implementation notes

On the parser level unique/structural were already optional, so it was pretty straightforward.

## Interesting/controversial decisions

None

## Test coverage

Updated tests

## Loose ends

None

unorsk committed:
automatically run ormolu

unorsk committed:
Merge branch 'unisonweb:trunk' into trunk

unorsk committed:
automatically run ormolu

unorsk committed:
use unique as default type modifier (when no modifier specified)

unorsk committed:
Fixing test for type modifiers, updating trnscripts

unorsk committed:
Merge branch 'unisonweb:trunk' into trunk

unorsk committed:
Merge remote-tracking branch 'origin/trunk' into unorsk/unique-as-default-type-modifier

aryairani commented on parser-typechecker/src/Unison/Syntax/DeclPrinter.hs:
```diff
@@ -199,8 +199,7 @@ fieldNames env r name dd = do
 
 prettyModifier :: DD.Modifier -> Pretty SyntaxText
 prettyModifier DD.Structural = fmt S.DataTypeModifier "structural"
-prettyModifier (DD.Unique _uid) =
-  fmt S.DataTypeModifier "unique" -- <> ("[" <> P.text uid <> "] ")
```
Could you leave in this comment so we don't forget the syntax for uid :) 😅

unorsk replied:
@aryairani yeah, sure! Done :)

unorsk committed:
A comment for pretty-printing unique types with uid

aryairani commented:
@unorsk With this PR, does it still optionally accept `unique`?
Edit: Yes it seems to.
<img width="236" alt="image" src="https://github.com/unisonweb/unison/assets/538571/6fbf1e53-d5f8-4f11-b6b2-6af85f141f45">
<img width="671" alt="image" src="https://github.com/unisonweb/unison/assets/538571/34fec8b2-293c-4a4f-8485-2ba6fbbfa28f">

Commit:
Merge branch 'trunk' into unorsk/unique-as-default-type-modifier

# Conflicts:
#	unison-src/transcripts/diff-namespace.output.md
#	unison-src/transcripts/docs.output.md

Commit:
update type-modifier-are-optional transcript

Commit:
fix fix1696 and fix2297 transcripts

these had broken in the past when we made the modifier mandatory
and nobody noticed until now

Commit:
fix fix2297

this had broken in the past when we made discarding non-Unit values explicit
and nobody noticed until now

Commit:
whitespace changes in transcripts??
